lib/ansible/modules/system/user.py

```diff
@@ -1813,6 +1866,24 @@ def modify_user_usermod(self):
 
         return (rc, out, err)
 
+    def user_info(self):
+        info = super(SunOS, self).user_info()
+        if info:
+            info += self._user_attr_info()
+        return info
+
+    def _user_attr_info(self):
+        info = [''] * 3
+        with open(self.USER_ATTR, 'r') as file_handler:
+            for line in file_handler:
+                lines = line.strip().split('::::')
+                if lines[0] == self.user:
+                    tmp = dict([x.split('=') for x in lines[1].split(';')])
```

Comment:
You can omit the `[]` here to use a generator expression, which is more efficient since the list is not created in memory. See [PEP 289](https://www.python.org/dev/peps/pep-0289/#rationale) for rationale.